source/modulo_components/include/modulo_components/utilities/utilities.hpp

```diff
@@ -61,4 +61,42 @@ parse_node_name(const rclcpp::NodeOptions& options, const std::string& fallback
   }
   return output;
 }
+
+/**
+ * @brief Modify parameter overrides to handle the rate and period parameters
+ * @details This function checks for existence of the rate and period parameter in the parameter overrides
+ * and modifies them to correspond to the same value. If both the rate and the period parameter exist, the period will
+ * be set from the rate.
+ * @param options The node options passed to the component constructor
+ * @return The same node options with modified parameter overrides
+*/
+[[maybe_unused]] static rclcpp::NodeOptions modify_parameter_overrides(const rclcpp::NodeOptions& options) {
+  auto modified = options;
+  rclcpp::Parameter rate;
+  rclcpp::Parameter period;
+  std::vector<rclcpp::Parameter> parameters;
+  for (const auto& parameter : options.parameter_overrides()) {
+    if (parameter.get_name() == "rate") {
+      rate = parameter;
+    } else if (parameter.get_name() == "period") {
+      period = parameter;
+    } else {
+      parameters.push_back(parameter);
+    }
+  }
+  if (rate.get_type() != rclcpp::ParameterType::PARAMETER_NOT_SET) {
+    rate = rclcpp::Parameter("rate", rate.as_int());
```

Comment:
is this line necessary? (same goes for the period assignment case a few lines down)

Reply:
I guess not strictly necessary no, more of a sanity thing to make sure they are indeed a parameter of the correct type.